Client/mods/deathmatch/logic/CPacketHandler.cpp

```diff
@@ -4359,8 +4359,13 @@ void CPacketHandler::Packet_ExplosionSync(NetBitStreamInterface& bitStream)
                 // Make sure the vehicle's blown
                 CClientVehicle* pExplodingVehicle = static_cast<CClientVehicle*>(pOrigin);
                 pExplodingVehicle->Blow(false);
+                if (!pExplodingVehicle->GetWillExplode())
+                    bCancelExplosion = true;
+                // Reset "will explode"
+                pExplodingVehicle->SetWillExplode(true);
 
                 // Call onClientVehicleExplode
+                // Bonus: This way "onClientVehicleExplode will only get called for serversided vehicles.
```

Comment:
Shouldn't we expect clientsided vehicles to trigger `onClientVehicleExplode`?

Reply:
I would expect it !

Reply:
That is the current way, only wanted to comment that.
Currently onClientVehicleExplode will only get triggered when the server sends the explosion.
You can try it on the current build.